lib/std/Thread/StaticResetEvent.zig

```diff
@@ -114,7 +114,7 @@ pub const DebugEvent = struct {
 };
 
 pub const AtomicEvent = struct {
```

Comment:
This struct makes me sad and is what prompted a [sync primitive rewrite](https://github.com/kprotty/zig/tree/posix-simple/lib/std/Thread). Tracking the waiters for set() was to account for window's NtKeyedEvent's API only waking one waiter at a time. `std.Thread.Futex` doesn't use that now so it can be made simpler. 

Trying to track waiters while using Futex isn't ideal as the value it waits on needs to stay the same until the set() happens. Each new thread wait()'ing bumps the waiters WAIT count potentially ending up in the waiting threads spuriously waking up to now wait on the new waiters count.

Either ways, a quick fix should be down below.